### PR TITLE
feat(wallet): adding all account address functions

### PR DESCRIPTION
**Pull request by kehiy**

### CODE REVIEW

File: wallet/vault/vault.go [hunk 1]
```diff
@@ -232,6 +232,21 @@ func (v *Vault) AllValidatorAddresses() []AddressInfo {
 	return addrs
 }
 
+func (v *Vault) AllAccountAddresses() []AddressInfo {
+	addrs := make([]AddressInfo, 0, v.AddressCount()/2)
+	for _, addrInfo := range v.Addresses {
+		addrPath, _ := addresspath.NewPathFromString(addrInfo.Path)
+		if addrPath.AddressType() == H(crypto.AddressTypeBLSAccount) {
```
Comment (b00f) [inline wallet/vault/vault.go:239]:
```suggestion
		if addrPath.AddressType() != H(crypto.AddressTypeValidator) {
```

This is more accurate. In future we may support other curves.

Comment (b00f) [reply wallet/vault/vault.go:239]: Add a test for this function, in that tests the return addresses has no validator address

File: wallet/wallet.go [hunk 2]
```diff
@@ -502,3 +506,24 @@ func (w *Wallet) AddTransaction(id tx.ID) error {
 func (w *Wallet) GetHistory(addr string) []HistoryInfo {
 	return w.store.History.getAddrHistory(addr)
 }
+
+func (w *Wallet) AllAmount(addr string, pt payload.Type) int64 {
```
Comment (b00f) [inline wallet/wallet.go:510]: Two notes:
1- The function name can change to "TotalAmount"
2- Let's add this in another PR. I think we may update the logic in a way to reduce the wasted changes.

**Review by b00f — CHANGES_REQUESTED** [review]

File: wallet/wallet.go (hunk 2 above)
Comment (kehiy) [reply wallet/wallet.go:510]: OK.

File: wallet/vault/vault_test.go [hunk 3]
```diff
@@ -123,6 +123,23 @@ func TestSortAddressInfo(t *testing.T) {
...
 }
 
+func TestAllAccountAddresses(t *testing.T) {
+	td := setup(t)
+
+	assert.Equal(t, td.vault.AddressCount(), 6)
+
+	accountAddrs := td.vault.AllAccountAddresses()
+	for _, i := range accountAddrs {
+		info := td.vault.AddressInfo(i.Address)
```
Comment (b00f) [inline wallet/vault/vault_test.go:133]: can be simplified:
```
assert.NotEqual(t, H(crypto.AddressTypeValidator), addresspath.NewPathFromString(addrInfo.Path).AddressType()
```

Also, please rename `NewPathFromString` to `FromString`

**Review by b00f — APPROVED** [review]

**Pull request merged**